cetmix_tower_server/models/cx_tower_plan_line.py

```diff
@@ -187,3 +191,32 @@ def _skip(self, server, plan_log_record, **kwargs):
             is_skipped=True,
             **log_vals,
         )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        PlanModel = self.env["cx.tower.plan"]
+
+        # Extract all unique plan_ids from vals_list
+        plan_ids = {vals.get("plan_id") for vals in vals_list if vals.get("plan_id")}
+
+        # Fetch all plan references in a single query
+        plans = PlanModel.browse(plan_ids)
+        plan_refs = {plan.id: plan.reference for plan in plans}
+
+        # Initialize line indices for each plan_id
+        plan_line_index_dict = defaultdict(int)
+
+        # Populate vals with references
+        for vals in vals_list:
+            plan_id = vals.get("plan_id")
+            if plan_id:
+                plan_line_index_dict[plan_id] += 1
+                line_index = plan_line_index_dict[plan_id]
+                vals["reference"] = f"{plan_refs[plan_id]}_{line_index}"
+            else:
+                # Handle the case where plan_id is not present
+                tmp_line_index = self._get_current_microsecond()
```

Comment:
Thanks
You are right, but when we run the tests, [this function](https://github.com/cetmix/cetmix-tower/blob/9e8617fcdb7aa7c1c5cbbdbf193d9e5caf208ee8/cetmix_tower_server/models/cx_tower_reference_mixin.py#L42) does not work correctly
yes, it checks for the presence of such a value in the table and shows that it is unique, but still, when creating a record, the sql_constraints is triggered and the creation of the record is blocked

and because of this I had to use such a function